src/Common/src/TypeSystem/Ecma/EcmaType.cs

```diff
@@ -537,6 +537,85 @@ private MarshalAsDescriptor GetMarshalAsDescriptor(FieldDefinition fieldDefiniti
             return null;
         }
 
+        public override PInvokeAttributes GetDelegatePInvokeAttributes()
+        {
+            PInvokeAttributes attributes = PInvokeAttributes.None;
+            if (!this.IsDelegate)
+            {
+                return attributes;
+            }
+
+            var customAttributeValue = this.GetDecodedCustomAttribute(
+                               "System.Runtime.InteropServices", "UnmanagedFunctionPointerAttribute");
+
+            if (customAttributeValue == null)
+            {
+                return attributes;
+            }
+
+            if (!customAttributeValue.HasValue)
+            {
+                return attributes;
+            }
+
+            if (customAttributeValue.Value.FixedArguments.Length != 1)
+            {
+                return attributes;
```

Comment:
Should this be [WinApi](https://github.com/dotnet/corert/blob/f9076cd8ec77b5c6f82b82aed620b1079153680d/src/System.Private.CoreLib/src/System/Runtime/InteropServices/UnmanagedFunctionPointerAttribute.cs#L22)?

Also, we don't go down to processing named arguments in this case?